### PR TITLE
Add helper to get props for pages that use LayoutWithMenu component

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Extract fetching logic required for any page that renders inside `LayoutWithMenu`

- Implement new helper function on relevant pages

## Justify why they are needed

Avoid duplication and make it easier to use the layout component.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/components/LayoutWithMenu/getLayoutWithMenuProps.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { GLOBAL_STORY_PROP_NAME } from '@/services/storyblok/Storyblok.constant'
+import { isRoutingLocale } from '@/utils/l10n/localeUtils'
+import {
+  GLOBAL_PRODUCT_METADATA_PROP_NAME,
+  fetchGlobalProductMetadata,
+} from './fetchProductMetadata'
+
+export const getLayoutWithMenuProps = async (
+  context: GetServerSidePropsContext | GetStaticPropsContext,
+  defaultApolloClient?: ApolloClient<unknown>,
```
Comment (robinandeer) [inline apps/store/src/components/LayoutWithMenu/getLayoutWithMenuProps.ts:15]: If we already have initialized a client, we want to reuse it to not mess with the cache

**Review by gustaveen — APPROVED** [review]
Very clean! 
Added via Giphy

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2740).

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2740) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2740).

**Pull request merged**